app/controllers/posts_controller.rb

```diff
@@ -22,6 +36,42 @@ def create
     end
   end
 
+  def update
+    # remove blank tag value
+    # convert from tag_id to tag object
+    values = post_params
+    values[:tags].shift
+    values[:tags].map! { |tag| Tag.find(tag) }
```

Comment:
You can reuse these code line with private method, define a private method that will return params with tags something like (check feature logic carefully again)
```
private

def post_params_with_tags
    // Insert code here
end
```